google/ads/google_ads/client.py

```diff
@@ -308,24 +315,23 @@ def intercept_unary_unary(self, continuation, client_call_details, request):
 
         Overrides abstract method defined in grpc.UnaryUnaryClientInterceptor.
 
+        Returns:
+            A grpc.Call instance representing a service response.
+
         Raises:
             GoogleAdsException: If the exception's trailing metadata
                 indicates that it is a GoogleAdsException.
             RpcError: If the exception's trailing metadata is empty or is not
                 indicative of a GoogleAdsException, or if the exception has a
                 status code of INTERNAL or RESOURCE_EXHAUSTED.
         """
-        try:
-            response = continuation(client_call_details, request)
-        except grpc.RpcError as ex:
-            self._handle_grpc_exception(ex)
-
-        if response.exception():
-            # Any exception raised within the continuation function that is not
-            # an RpcError will be set on the response object and raised here.
-            raise response.exception()
+        response = continuation(client_call_details, request)
+        exception = response.exception()
 
-        return response
+        if exception:
+            self._handle_grpc_failure(exception)
```

Comment:
This should be `self._handle_grpc_failure(response)` (the `response` is the `Call` instance, and `exception` may be of a different, e.g., user-generated, type)

Reply:
Other than this, the changes in use of gRPC code here LGTM

Reply:
So for most Rpc errors `response` and `response.exception()` will in fact be the same object. Does that include errors that are generated by the service a request is made to? For instance, if a request is mal-constructed and the response is a 400 error? 

If so then any exception that isn't `RpcError` is likely originating from somewhere in _this library_?

Reply:
FYI Here's the commit where I address this: 8d509b3b9faf805415da9f0b543ff632a37ece05

Reply:
I'm not sure without digging further into our code; the only obvious path for such an exception to be raised would be if your interceptor implementation raised one itself, but `_handle_grpc_failure` would in turn throw an exception if that occurred and its argument were not a `Call` (since it invokes its `code()` method).